federation/pkg/dnsprovider/providers/coredns/rrsets.go

```diff
@@ -78,13 +77,10 @@ func (rrsets ResourceRecordSets) Get(name string) (dnsprovider.ResourceRecordSet
 		}
```

Comment:
@miekg, we are still using etcd2 client for this library (same as CoreDNS). we would plan and upgrade to use etcd3 client ASAP. I think CoreDNS should be first upgraded to use etcd3 client then followed by this library.
@csbell, i would add detecting other types from etcd records in a follow-up pr.